src/tree-manager/flatten-tree.js

```diff
@@ -92,31 +92,36 @@ const tree = [
 }
 ```
  * @param  {[type]} tree              The incoming tree object
- * @param  {[bool]} simple            Whether its in Single slect mode (simple dropdown)
+ * @param  {[bool]} simple            Whether its in Single select mode (simple dropdown)
+ * @param  {[bool]} radio             Whether its in Radio select mode (radio dropdown)
  * @param  {[bool]} showPartialState  Whether to show partially checked state
  * @return {object}                   The flattened list
  */
-function flattenTree(tree, simple, showPartialState, hierarchical) {
+function flattenTree({ tree, simple, radio, showPartialState, hierarchical }) {
   const forest = Array.isArray(tree) ? tree : [tree]
 
   // eslint-disable-next-line no-use-before-define
-  const { list, defaultValues } = walkNodes({
+  const { list, defaultNodes, checkedNodes } = walkNodes({
     nodes: forest,
     simple,
+    radio,
     showPartialState,
     hierarchical
   })
-  return { list, defaultValues }
+  const defaultValues = defaultNodes.map(i => i._id)
+  const checkedValues = checkedNodes.map(i => i._id)
```

Comment:
Can we return the `ids` while walking so that we can avoid these 2 extra iterations?

Reply:
The reason I internally keep track of the nodes instead of ids is because I need to uncheck default values if finding a checked first. If I moved that to the return part it would be even more iterations depending on number of levels/nodes processed.

But you have different ideas about what should be checked first in the case of single select so perhaps that is not needed anymore.

Reply:
Just trying to avoid extra iterations if possible.

Reply:
I would have to add an if(depth === 0) and do the same iteration in walkNodes either way to avoid even more iterations. Unless we split upp the walking/traversing this is better since less checks are done in this way.

Reply:
I've added a benchmark here, it's very random "who wins" but I cannot see that the new iteration makes a big difference at least:
https://ellinge.github.io/react-dropdown-tree-select-test/
![bild](https://user-images.githubusercontent.com/17863113/55038199-5c58a680-5020-11e9-93f5-142798666793.png)
![bild](https://user-images.githubusercontent.com/17863113/55038212-67133b80-5020-11e9-944d-1bf73f85a8b1.png)

Reply:
IE11 is much slower though. Not sure if it's related to flattenTree or something else though:
![bild](https://user-images.githubusercontent.com/17863113/55039116-41d3fc80-5023-11e9-8b1f-a503fdbf845b.png)

Reply:
Let me take a look and I'll get back to you.

I wouldn't worry about IE 11, the slowness is expected.

Reply:
After last commit it performs well in IE11 as well